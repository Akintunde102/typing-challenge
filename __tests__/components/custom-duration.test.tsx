import { fireEvent, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import CustomDuration from '../../components/custom-duration';

describe('Custom Duration Component', () => {
    const mockHandleDurationPick = jest.fn();

    describe("when showCustomDuration is set to  true", () => {
        let renderedComponent: any;

        beforeEach(() => {
            renderedComponent = render(<CustomDuration showCustomDuration={true} handleDurationPick={mockHandleDurationPick} />)
        })

        it('renders a text', () => {
            const text = screen.getByText("Fill In Custom (in mins)");
            expect(text).toBeInTheDocument()
        })

        it('has a link to set duration', () => {
            const button = screen.getByText('Set').closest('button');

            fireEvent.click(button)

            expect(mockHandleDurationPick).toHaveBeenCalledTimes(1);
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        })
    })


    describe("when showCustomDuration is set to false", () => {
        let renderedComponent: any;

        beforeEach(() => {
            renderedComponent = render(<CustomDuration showCustomDuration={false} handleDurationPick={mockHandleDurationPick} />)
        })

        it('does not renders a text', () => {
            expect(renderedComponent.container).toBeEmptyDOMElement();
        })

        it('matches snapshot', () => {
            expect(renderedComponent).toMatchSnapshot()
        })
    })
})