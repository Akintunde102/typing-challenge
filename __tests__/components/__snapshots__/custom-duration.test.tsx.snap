// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Duration Component when showCustomDuration is set to  true matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="customDuration"
      >
        <div
          class="smallForm"
        >
          <div>
            Fill In Custom (in mins)
          </div>
          <div
            class="bigInput"
          >
            <div>
              <input
                size="5"
                value="1"
              />
            </div>
            <button>
              Set
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="customDuration"
    >
      <div
        class="smallForm"
      >
        <div>
          Fill In Custom (in mins)
        </div>
        <div
          class="bigInput"
        >
          <div>
            <input
              size="5"
              value="1"
            />
          </div>
          <button>
            Set
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Custom Duration Component when showCustomDuration is set to false matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
